src/xunit.analyzers/Descriptors.cs

```diff
@@ -254,7 +254,10 @@ internal static class Descriptors
             "Keep precision in range {0} when asserting equality of {1} typed actual value.",
             Categories.Assertions, DiagnosticSeverity.Error, isEnabledByDefault: true);
 
-        // Placeholder for rule X2017
+        internal static DiagnosticDescriptor X2017_AssertIsTypeShouldNotBeUsedForAbstractType { get; } = new DiagnosticDescriptor("xUnit2017",
+            "Do not assert that an object's exact type is an abstract class or interface",
+            "Do not assert that an object's exact type is the {0} '{1}'.",
+            Categories.Assertions, DiagnosticSeverity.Error, isEnabledByDefault: true);
```

Comment:
Since asserting on an interface or abstract class does not actually cause the test to blow up, this should be a warning, not an error.